### DISCUSSION TITLE
Attach Pylenium to an existing browser

### DISCUSSION THREAD

**Discussion by serdnago** [ElSnoMan/pyleniumio#193] [Q&A] [answered]
Is it possible to attach Pylenium to an existing browser session? We have some tests that require mocking some calls (using selenium-wire). The tests start the browser, hit some URLs, mock some of the requests/responses and then proceed to regular UI testing. 
Thanks in advance and congratulations for the fantastic framework!

**Comment by ElSnoMan** [comment, marked as answer]
With Selenium, you pass in the debug port of the existing browser session:

```java
chromeOptions options = new ChromeOptions();
options.setExperimentalOption("debuggerAddress", "127.0.0.1:1111");
WebDriver driver = new ChromeDriver(options);
```

So, with Pylenium, once you have the debug port, you can either use `pylenium.json` to have _all_ tests run against the current browser by setting the [Experimental Options as seen in the docs](https://docs.pylenium.io/configuration/driver#experimental_options), or you can control it at just the test level:

```python
def test_example(py: Pylenium):
    py.config.driver.experimental_options = {"debuggerAddress": 12345}
    # you are now on the existing browser session
    ...
```

**Reply by serdnago** [reply]
Works like a charm! Thank you so much!
